### PR TITLE
Add BGS/VACOLS latency metrics

### PR DESCRIPTION
**Pull request by joofsh**
Sets the proper service & name for all BGS and VACOLS requests

Test Plan:
- [x] Manually run BGS requests from console
- [x] Manually run VACOLS queries from console

### CODE REVIEW

File: app/services/appeal_repository.rb [hunk 1]
```diff
@@ -7,9 +7,9 @@ def log(event, data)
     when :request
       status = data[:response_code]
       name = data[:request].class.name.split("::").last
-      application = RequestStore[:application] || "other"
+      app = RequestStore[:application] || "other"
       PrometheusService.completed_vbms_requests.increment(status: status,
-                                                          application: application,
+                                                          app: app,
```
Comment (joofsh) [inline app/services/appeal_repository.rb]: this was a typo from previous PRs. We've called it `app` elsewhere

File: app/services/appeal_repository.rb [hunk 2]
```diff
@@ -173,10 +181,7 @@ def self.establish_claim!(appeal:, claim:)
     @vbms_client ||= init_vbms_client
 
     sanitized_id = appeal.sanitized_vbms_id
-
-    raw_veteran_record = MetricsService.timer "BGS: fetch_veteran_info #{appeal.vacols_id}" do
-      BGSService.new.fetch_veteran_info(sanitized_id)
-    end
+    raw_veteran_record = BGSService.new.fetch_veteran_info(sanitized_id)
```
Comment (joofsh) [inline app/services/appeal_repository.rb]: Internal to `BGSService` we were already wrapping every call in MetricService. So this was double logging events

File: app/services/appeal_repository.rb [hunk 3]
```diff
@@ -116,15 +120,19 @@ def self.issues(vacols_id:)
 
   # :nocov:
   def self.remands_ready_for_claims_establishment
-    remands = MetricsService.timer "loaded remands in loc 97 from VACOLS" do
+    remands = MetricsService.timer("VACOLS: remands_ready_for_claims_establishment",
+                                   service: :vacols,
+                                   name: "remands_ready_for_claims_establishment") do
```
Comment (joofsh) [inline app/services/appeal_repository.rb]: So for BGS & VBMS, we're setting the `name` to be the API request we make (e.g. `ListDocuments`), VACOLS has way more permutations of "requests", so instead we'll name it by the method calling it. I think this is the easiest way to trace back a specific VACOLS query to where it originates from in the code

Comment (askldjd) [inline app/services/appeal_repository.rb]: Ya, this make sense.

File: app/services/appeal_repository.rb [hunk 4]
```diff
@@ -191,7 +196,7 @@ def self.update_vacols_after_dispatch!(appeal:, vacols_note: nil)
       update_location_after_dispatch!(appeal)
 
       if vacols_note
-        VACOLS::Note.create!(case_record: appeal.case_record,
+       VACOLS::Note.create!(case_record: appeal.case_record,
```
Comment (askldjd) [inline app/services/appeal_repository.rb]: We lost a space :)

File: config/initializers/warmup_vacols.rb [hunk 5]
```diff
@@ -24,7 +24,10 @@
   Rails.logger.info("creating #{initial_pool_size} initial connections...")
 
   unless ApplicationController.dependencies_faked?
-    MetricsService.timer "created #{initial_pool_size} connections" do
+    MetricsService.timer("VACOLS: warmup_vacols #{initial_pool_size} connections",
```
Comment (askldjd) [inline config/initializers/warmup_vacols.rb]: So we will probably only get one sample of this for each deploy. Is that right?

Comment (joofsh) [inline config/initializers/warmup_vacols.rb]: correct

File: app/services/appeal_repository.rb (hunk 2 above)
Comment (amprokop) [inline app/services/appeal_repository.rb]: 👍

File: app/services/appeal_repository.rb (hunk 3 above)
Comment (amprokop) [inline app/services/appeal_repository.rb]: 👍

**Pull request merged**